src/main/kotlin/com/sourcegraph/cody/chat/ChatSession.kt

```diff
@@ -59,6 +59,8 @@ private constructor(private val project: Project, newSessionId: CompletableFutur
 
   private val logger = LoggerFactory.getLogger(ChatSession::class.java)
 
+  private val theMostRecentContextFilesMessageId = AtomicReference<UUID>(UUID.randomUUID())
```

Comment:
I don't like that state variable. It is used for updating single message context but it pollutes whole class. We should think about:
a) removing it completely somehow
b) if not possible then maybe moving to single message - then you don't need fake start UUID, just a boolean